sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/SchemaPruningSuite.scala

```diff
@@ -338,6 +340,75 @@ abstract class SchemaPruningSuite
     }
   }
 
+  testSchemaPruning("select one deep nested complex field after repartition") {
```

Comment:
Is this test related to this PR?

Reply:
Not for this change, but we don't have test case for repartition. Added one to improve test coverage.

Reply:
Ah, ok.